modules/usage/module.php

```diff
@@ -564,22 +578,29 @@ private function save_document_usage( Document $document ) {
 		// Get data manually to avoid conflict with `\Elementor\Core\Base\Document::get_elements_data... convert_to_elementor`.
 		$data = $document->get_json_meta( '_elementor_data' );
 
-		if ( ! empty( $data ) ) {
-			try {
-				$usage = $this->get_elements_usage( $document->get_elements_raw_data( $data ) );
+		try {
+			$this->add_document_usage( $document, $data );
+		} catch ( \Exception $exception ) {
+			return; // Do nothing.
+		};
+	}
 
-				$document->update_meta( self::META_KEY, $usage );
+	private function add_document_usage( $document, $data ) {
+		if ( $data ) {
+			$elements_usage = $this->get_elements_usage( $document->get_elements_raw_data( $data ) );
```

Comment:
Move `get_elements_usage` to the document itself.

Reply:
It's a good idea, please make it in the Controls_Stack level, in order to use it also for elements

Reply:
Handled.